libwebcash/src/webcash.c

```diff
@@ -13,67 +13,93 @@
 #include "sha256.h"
 #include "webcash.h"
 
+// Common to all database records, the id is the primary key and the timestamp
+// records when the record was created.
 struct webcash_database_key {
     size_t id;
     time_t timestamp;
 };
 
+// Accepting the terms of service is recorded by storing a record with the text
+// of the terms of service, and implicitly the time of acceptance from the
+// dbkey.timestamp field.
 struct webcash_terms {
     struct webcash_database_key dbkey;
     char* body;
 };
 
+// A secret is an unguessable value that can be used to redeem webcash.  It can
+// be generated by the wallet code using the wallet's master secret, or it can
+// explicitly imported.  All secrets have two boolean classifications:
+//
+// "mine" indicates that this wallet is the sole posessor of the secret.
+//
+// "sweep" indicates that this wallet should redeem the secret as soon as it is
+// seen, rather than waiting for the user to implicitly redeem it as part of a
+// replacement.
 struct webcash_wallet_secret {
-    struct webcash_wallet_secret *next;
+    struct webcash_wallet_secret *next; // linked list pointer
```

Comment:
`T* x` or `T *x`. I prefer the former, but I can live with either; let's stick to one. :)